src/blocks/donate/view.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Server-side rendering of the `newspack-blocks/donate` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `newspack-blocks/donate` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string
+ */
+function newspack_blocks_render_block_donate( $attributes ) {
+	if ( ! class_exists( 'Newspack\Configuration_Managers' ) ) {
+		return '';
+	}
+
+	Newspack_Blocks::enqueue_view_assets( 'donate' );
+
+	$configuration_manager = Newspack\Configuration_Managers::configuration_manager_class_for_plugin_slug( 'woocommerce' );
+	$settings = $configuration_manager->get_donation_settings();
+	if ( is_wp_error( $settings ) || ! $settings['created'] ) {
+		return '';
+	}
+
+	$frequencies = [
+		'once'  => __( 'One-time', 'newspack' ),
+		'month' => __( 'Monthly', 'newspack' ),
+		'year'  => __( 'Annually', 'newspack' ),
+	];
+
+	$selected_frequency = 'month';
+	$suggested_amounts  = $settings['suggestedAmounts'];
+
+	ob_start();
+
+	/**
+	 * For AMP-compatibility, the donation forms are implemented as pure HTML forms (no JS).
+	 * Each frequency and tier option is a radio input, styled to look like a button.
+	 * As the radio inputs are checked/unchecked, fields are hidden/displayed using only CSS.
+	 */
+
+	if ( ! $settings['tiered'] ) :
+
+		?>
+		<div class='newspack-blocks-donate untiered'>
+			<form>
+				<input type='hidden' name='newspack_donate' value='1' />
+				<div class='newspack-blocks-donate__options'>
+					<?php foreach ( $frequencies as $frequency_slug => $frequency_name ) : ?>
+						<?php $esc_slug = esc_attr( $frequency_slug ); ?>
+						<?php $amount = 'year' === $frequency_slug ? 12 * $settings['suggestedAmountUntiered'] : $settings['suggestedAmountUntiered']; ?>
+
+						<div class='newspack-blocks-donate__frequency'>
+							<input type='radio' value='<?php echo $esc_slug; ?>' id='newspack-donate-<?php echo $esc_slug; ?>' name='donation_frequency' <?php checked( $selected_frequency, $frequency_slug ); ?> />
```

Comment:
Tiny nitpick/recent learning... It was recently suggested to me that it's better to do separate escaping for every attribute, even if it means duplicate code. This makes it slightly easier to review, as the reviewer can simply examine every single attribute and verify correct escaping.

```suggestion
							<input type='radio' value='<?php echo esc_attr( $frequency_slug ); ?>' id='newspack-donate-<?php echo esc_attr( $frequency_slug ); ?>' name='donation_frequency' <?php checked( $selected_frequency, $frequency_slug ); ?> />
```

Reply:
Makes sense to me. Done in https://github.com/Automattic/newspack-blocks/pull/64/commits/2059d7f80819b60d3a34a153e6903bfede42a89b along with some re-arranging to make things readable.